server/src/main/java/org/apache/cassandra/sidecar/lifecycle/ProcessLifecycleProvider.java

```diff
@@ -18,38 +18,232 @@
     }
 
     @Override
     public boolean isRunning(InstanceMetadata instance)
     {
-        throw new UnsupportedOperationException("Not implemented yet");
+        return isCassandraProcessRunning(instance);
+    }
+
+    private void startCassandra(InstanceMetadata instance)
+    {
+        ProcessRuntimeConfiguration runtimeConfig = getRuntimeConfiguration(instance);
+        try
+        {
+            String stdoutLocation = getStdoutLocation(runtimeConfig.instanceName());
+            String stderrLocation = getStderrLocation(runtimeConfig.instanceName());
+            String pidFileLocation = getPidFileLocation(runtimeConfig.instanceName());
+            ProcessBuilder processBuilder = runtimeConfig.buildStartCommand(pidFileLocation,
+                                                                            stdoutLocation,
+                                                                            stderrLocation);
+            LOG.info("Starting Cassandra instance {} with command: {}", runtimeConfig.instanceName(), processBuilder.command());
+
+            Process process = processBuilder.start();
+            process.waitFor(CASSANDRA_PROCESS_TIMEOUT_MS, TimeUnit.MILLISECONDS); // blocking call, make async?
+
+            if (isCassandraProcessRunning(instance))
+            {
+                LOG.info("Started Cassandra instance {} with PID {}", runtimeConfig.instanceName(), readPidFromFile(Path.of(pidFileLocation)));
+            }
+            else
+            {
+                throw new RuntimeException("Failed to start Cassandra instance " + runtimeConfig.instanceName() +
+                                           ". Check stdout at " + stdoutLocation + " and stderr at " + stderrLocation);
+            }
+        }
+        catch (Throwable t)
+        {
+            throw new RuntimeException("Failed to start Cassandra instance " + runtimeConfig.instanceName() + " due to " + t.getMessage(), t);
+        }
+    }
+
+    private void stopCassandra(InstanceMetadata instance)
+    {
+        ProcessRuntimeConfiguration casCfg = getRuntimeConfiguration(instance);
+        try
+        {
+            String pidFileLocation = getPidFileLocation(casCfg.instanceName());
+            Long pid = readPidFromFile(Path.of(pidFileLocation));
+            Optional<ProcessHandle> processHandle = ProcessHandle.of(pid);
+            if (processHandle.isPresent())
+            {
+                LOG.info("Stopping process of Cassandra instance {} with PID {}.", casCfg.instanceName(), pid);
+                CompletableFuture<ProcessHandle> terminationFuture = processHandle.get().onExit();
+                processHandle.get().destroy();  // blocking call, make async?
+                terminationFuture.get(CASSANDRA_PROCESS_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+            }
+            else
+            {
+                LOG.warn("No process running for Cassandra instance {} with PID {}.", casCfg.instanceName, pid);
```

Comment:
Call the method for or consistency.
```suggestion
                LOG.warn("No process running for Cassandra instance {} with PID {}.", casCfg.instanceName(), pid);
```